scenedetect/video_manager.py

```diff
@@ -743,8 +743,13 @@ def read(self):
         if self._curr_cap is not None and self._end_of_video != True:
             while not read_frame:
                 read_frame, self._last_frame = self._curr_cap.read()
-                if not read_frame and not self._get_next_cap():
-                    break
+                # Switch to the next capture when the current one is over
+                if not read_frame:
+                    # Break the loop when all the captures are over
+                    if not self._get_next_cap():
+                        break
+                    # Get frame of the new capture
+                    read_frame, self._last_frame = self._curr_cap.read()
```

Comment:
Replace this line (line 752) with 'continue' instead - I think that will achieve what I was aiming for in the first comment I made.

Reply:
`cv::VideoCapture.read()` [returns](https://docs.opencv.org/3.4/d8/dfe/classcv_1_1VideoCapture.html#a473055e77dd7faa4d26d686226b292c1) (False, None) in the case where a video doesn't have any frames for some reason. Therefore, I propose to switch from while to if as one read attempt would suffice to get the next frame:
```suggestion
if self._curr_cap is not None and self._end_of_video != True:
    read_frame, self._last_frame = self._curr_cap.read()

    # Switch to the next capture when the current one is over
    if not read_frame and self._get_next_cap():
        read_frame, self._last_frame = self._curr_cap.read()

    # Downscale frame if there was any
    if read_frame and self._downscale_factor > 1:
        self._last_frame = self._last_frame[
            ::self._downscale_factor, ::self._downscale_factor, :]
```

Reply:
Ah good point!  Very nice refactor, looks great :)